storage/kubernetes/controller.go

```diff
@@ -170,36 +171,46 @@ func (s *storage) saveInK8s(secret *v1.Secret) (*v1.Secret, error) {
 		}
 	}
 
+	// ensure that the merged secret actually contains data before overwriting the existing fields
+	if len(secret.Data) < 2 {
```

Comment:
Nit: Can we make "2" a constant with a descriptive name? Can be done someother time given the context of this change.

Reply:
Yeah, all this checking of data len isn't great. Added a more explicit check that mirrors what upstream is doing at https://github.com/kubernetes/kubernetes/blob/dcce2357ffda4b8cc132296a1e5a8d74bf9b150b/pkg/apis/core/validation/validation.go#L5577-L5583